### PR TITLE
DocDB: moving create to AsyncCollector; improving create logic

### PR DESCRIPTION
**Pull request by brettsam**

### CODE REVIEW

File: src/WebJobs.Extensions.DocumentDB/DocumentDBUtility.cs [hunk 1]
```diff
@@ -51,8 +37,10 @@ internal static class DocumentDBUtility
 
                     if (TryGetDocumentClientException(ex, out de))
                     {
-                        if (de.StatusCode == HttpStatusCode.NotFound &&
-                            ignoreNotFound)
+                        if (de.StatusCode.HasValue &&
+                            codesToIgnore != null &&
+                            (int)de.StatusCode.Value != 429 &&
+                            codesToIgnore.Contains(de.StatusCode.Value))
```
Comment (pragnagopa) [inline src/WebJobs.Extensions.DocumentDB/DocumentDBUtility.cs]: simplify using linq?

File: test/WebJobs.Extensions.Tests/Extensions/DocumentDB/DocumentDBUtilityTests.cs [hunk 2]
```diff
@@ -20,27 +17,27 @@ namespace Microsoft.Azure.WebJobs.Extensions.Tests.Extensions.DocumentDB
...
+            }, 0, HttpStatusCode.Conflict, HttpStatusCode.NotFound);
 
             // Assert
             Assert.Equal(testString, s);
             // The fact that it doesn't throw proves that it was ignored
         }
 
         [Fact]
-        public async Task ExecuteAndIgnoreStatusCode_DoesNotIgnore_OtherStatusCodes()
+        public async Task RetryAsync_DoesNotIgnore_OtherStatusCodes()
```
Comment (pragnagopa) [inline test/WebJobs.Extensions.Tests/Extensions/DocumentDB/DocumentDBUtilityTests.cs]: Add Test StatusCodes null

**Comment by pragnagopa** [comment]
Couple of nits. Looks good :+1:

File: test/WebJobs.Extensions.Tests/Extensions/DocumentDB/DocumentDBUtilityTests.cs (hunk 2 above)
Comment (brettsam) [inline test/WebJobs.Extensions.Tests/Extensions/DocumentDB/DocumentDBUtilityTests.cs]: Good point. I've added a test for null.

---
In reply to: [60487790](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/83#discussion_r60487790) [](ancestors = 60487790)
